### DISCUSSION TITLE
How to set xref for new layers

### DISCUSSION THREAD

**Discussion by yingyue4008** [LibreDWG/libredwg#614] [Q&A] [answered]
Excuse me, I found xref is empty after creating a new layer, and then I used dwg_add_handleref function is assigned, but it is not known what value is passed in for the corresponding parameter of the function.
And I don't know if this is correct?

**Comment by rurban** [comment, marked as answer]
I don't think we do have an example yet of a layer->xref in the wild. You might need to create one, and look at the log.
It's a type 5 handle to an external XREF block. (blkisxref). See spec.h

**Reply by yingyue4008** [reply]
> I don't think we do have an example yet of a layer->xref in the wild. You might need to create one, and look at the log. It's a type 5 handle to an external XREF block. (blkisxref). See spec.h

Thank you very much. I tried to do it here

**Reply by yingyue4008** [reply]
pNewLayer->xref = dwgAddHandleref (&dwg, 5, 0, NULL);

How to set the values of the 3、4 parameters in this dwgAddHandleref()?

**Reply by rurban** [reply]
3rd is the handle value, 4th is usually not needed for code 5, but our obj

**Reply by yingyue4008** [reply]
> 3rd is the handle value, 4th is usually not needed for code 5, but our obj

Should I find a new handle function? Should not be used dwgAddHandleref () to xref

**Comment by yingyue4008** [comment]
unsigned long absref = dwg.object[dwg.num_objects -1].handle.value + 1;
pNewLayer->xref = dwgAddHandleref(dwg, 5, absref, NULL);
Excuse me,How to make the value(handle.value + 1) effective? Add a new obj?
